enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+---
+title: lb-subnet-selection-aws
+authors:
+  - "@gcs278"
+reviewers:
+  - "@candita"
+  - "@frobware"
+  - "@rfredette"
+  - "@alebedev87"
+  - "@miheer"
+  - "@JoelSpeed, for review regarding CCM"
+approvers:
+  - "@Miciah"
+api-approvers:
+  - "@deads2k"
+creation-date: 2024-03-06
+last-updated: 2024-04-10
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/NE-705
+see-also:
+replaces:
+superseded-by:
+---
+
+# LoadBalancer Subnet Selection for AWS
+
+## Summary
+
+This enhancement extends the IngressController API to allow a cluster admin to
+configure specific subnets for AWS load balancers. By default, AWS auto-discovers
+the subnets, with logic to break ties when there are multiple subnets for a particular
+availability zone. This proposal works by configuring the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation on the
+LoadBalancer-type service, followed by a deletion of the service at a time when
+disruption can be tolerated.
+
+In addition, this enhancement will also provide an install-time configuration for Day 1
+operations by extending the OpenShift install-config API. The install-config API will
+provide the ability to configure specific subnets for all IngressControllers, including
+configuring the default IngressController's subnets.
+
+## Motivation
+
+Cluster admins on AWS may have dedicated subnets for their load balancers due to
+security reasons, architecture, or infrastructure constraints. 
+
+Currently, cluster admins can configure subnets by manually adding the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation to the
+service exposing the IngressController. However, this approach does not
+allow deploy-time subnet configuration and is unsupported because the
+service is managed by Ingress Operator.
```

Comment:
nit: this doesn't really explain why it is unsupported.  Does this mean that Ingress Operator will revert the changes, or something else?